packages/terminal/src/browser/base/terminal-widget.ts

```diff
@@ -36,6 +40,13 @@ export abstract class TerminalWidget extends BaseWidget {
      */
     abstract sendText(text: string): void;
 
+    /**
+     * Build a full command line to send via stdin to the running shell.
+     *
+     * @param commandOptions - command, arguments and cwd.
+     */
+    abstract buildAndExecuteCommand(commandOptions: CommandLineOptions): Promise<void>;
```

Comment:
`build` bit seems to be implementation detail, the real intention is `executeCommand`